waf_unsupported_test.go

```diff
@@ -20,12 +19,16 @@ func TestLoad(t *testing.T) {
 	ok, err := waf.Load()
 	require.False(t, ok)
 	require.Error(t, err)
-	var expectedErr *waf.UnsupportedTargetError
-	require.Truef(t, errors.As(err, &expectedErr), "unexpected error of type %[1]T: %[1]v", err)
 }
 
 func TestSupportsTarget(t *testing.T) {
 	supported, err := waf.SupportsTarget()
 	require.False(t, supported)
 	require.Error(t, err)
 }
+
+func TestHealth(t *testing.T) {
+	ok, err := waf.Health()
+	require.False(t, ok)
+	require.Error(t, err)
```

Comment:
The last [big] mile of effort would be to test every single error type ;-p

Reply:
We would need to divise test files with different build tags.... meh.... (till we refactored go-libddwaf with more packages)

Reply:
lol yes that's why it's a hard one - but definitely worth it given this complex (but useful) error generation that you have built.

worth a TODO otherwise:
```suggestion
	// TODO: finer-grain error checking of all the different error possibilities (unsupported go version, unsupported target, disabled, etc.)
	require.Error(t, err)
```